core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java

```diff
@@ -78,20 +77,12 @@ public HttpResponse sendPostRequest(HttpPost post) throws IOException, FriendlyE
         try {
             return getHttpClient().execute(post);
         } catch (SSLHandshakeException e) {
-            String completeUrl = "https://"+post.getURI().getHost();
-            if(completeUrl.equals(Defaults.LIVE_ENDPOINT)) {
-                throw new FriendlyException("ApplicationInsights Java Agent failed to connect to Live metric end point.",
-                        "Unable to find valid certification path to requested target.",
-                        "Please import the SSL certificate from " + completeUrl + ", into the java key store. "+
-                        "Learn more about importing the certificate here: https://go.microsoft.com/fwlink/?linkid=2151450",
-                        "This message is only logged the first time it occurs after startup.");
-            } else {
-                throw new FriendlyException("ApplicationInsights Java Agent failed to send telemetry data.",
-                        "Unable to find valid certification path to requested target.",
-                        "Please import the SSL certificate from " + completeUrl + ", into the java key store. "+
-                        "Learn more about importing the certificate here: https://go.microsoft.com/fwlink/?linkid=2151450",
-                        "This message is only logged the first time it occurs after startup.");
-            }
+            String completeUrl = "https://" + post.getURI().getHost();
+            boolean isUsingCustomKeyStore = (System.getProperty("javax.net.ssl.trustStore") != null);
+            throw new FriendlyException(SSLUtil.getSSLFriendlyExceptionBanner(completeUrl),
+                    SSLUtil.getSSLFriendlyExceptionMessage(),
+                    SSLUtil.getSSLFriendlyExceptionAction(completeUrl, isUsingCustomKeyStore),
+                    SSLUtil.getSSLFriendlyExceptionNote());
```

Comment:
```suggestion
            SSLUtil.throwFriendlyException(completeUrl);
```

and then you can check `javax.net.ssl.trustStore` inside of `throwFriendlyException`